AppTaskQueue/appscale/taskqueue/queue.py

```diff
@@ -940,20 +943,26 @@ def _delete_index(self, eta, task_id):
                   'id': task_id}
     self.db_access.session.execute(delete_index, parameters)
 
-  def _delete_task_and_index(self, task):
+  def _delete_task_and_index(self, task, retries=5):
     """ Deletes a task and its index atomically.
 
     Args:
       task: A Task object.
     """
-    batch_delete = BatchStatement(retry_policy=BASIC_RETRIES)
-
     delete_task = SimpleStatement("""
       DELETE FROM pull_queue_tasks
       WHERE app = %(app)s AND queue = %(queue)s AND id = %(id)s
-    """)
+      IF EXISTS
+    """, retry_policy=NO_RETRIES)
     parameters = {'app': self.app, 'queue': self.name, 'id': task.id}
-    batch_delete.add(delete_task, parameters=parameters)
+    try:
+      self.db_access.session.execute(delete_task, parameters=parameters)
+    except TRANSIENT_CASSANDRA_ERRORS:
+      retries_left = retries - 1
+      if retries_left <= 0:
+        raise
+      logger.warning('Task deletion may have failed. Retrying.')
```

Comment:
I'm sorry for being nagging=) but would you mind adding string representation of an error?